Week7/app/src/main/java/com/em4n0101/mytvshows/ui/searchshow/SearchShowFragment.kt

```diff
@@ -0,0 +1,140 @@
+import com.em4n0101.mytvshows.MyTvShowsApplication
+import com.em4n0101.mytvshows.R
+import com.em4n0101.mytvshows.model.Success
+import com.em4n0101.mytvshows.model.Show
+import com.em4n0101.mytvshows.networking.NetworkingStatusChecker
+import com.em4n0101.mytvshows.ui.showdetail.ShowDetailActivity
+import com.em4n0101.mytvshows.utils.toast
+import kotlinx.android.synthetic.main.fragment_search_show.*
+import kotlinx.coroutines.launch
+
+class SearchShowFragment : Fragment() {
+
+    private val remoteApi = MyTvShowsApplication.remoteApi
+    private val networkStatusChecker by lazy {
+        NetworkingStatusChecker(activity?.getSystemService(ConnectivityManager::class.java))
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_search_show, container, false)
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        activity?.let {
+            // setup recycler
+            if (context?.resources?.configuration?.orientation == Configuration.ORIENTATION_PORTRAIT) {
+                showsRecyclerView.layoutManager = GridLayoutManager(context, 2)
+            } else {
+                showsRecyclerView.layoutManager = GridLayoutManager(context, 4)
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.search_menu, menu)
+        val searchItem = menu.findItem(R.id.actionSearchShow)
+        if (searchItem != null) {
+            val searchView = searchItem.actionView as SearchView
+            val editText = searchView.findViewById<EditText>(R.id.search_src_text)
+            editText.hint = getString(R.string.hintSearchShow)
+
+            searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+                override fun onQueryTextSubmit(query: String?): Boolean {
+                   query?.let {
+                        if (it.isNotBlank()) searchFor(it)
+                    }
+                    return true
+                }
+
+                override fun onQueryTextChange(newText: String?): Boolean {
+                    val minCharactersToSearch = 3
+                    newText?.let {
+                        if (it.length >= minCharactersToSearch) {
+                            searchFor(it)
+                        }
+                        else {
+                            updateUiWithShowList(emptyList())
+                        }
+                    }
+                    return true
+                }
+            })
+        }
+        super.onCreateOptionsMenu(menu, inflater)
```

Comment:
This is a bit too long in my opinion. You can create extensions for textchangeListener and pass in lambdas that will run when needed.